src/IW/Effects/Cabal.hs

```diff
@@ -0,0 +1,61 @@
+{- | This module contains the class definitiion of @MonadCabal@ and
+an instance of @MonadCabald@ for the @App@ monad. Instances of
+@MonadCabal@ have a @getCabalCategories@ action that returns @[Category]@
+given a @RepoOwner@ and @RepoName@.
+-}
+
+module IW.Effects.Cabal
+       ( getCabalCategories
+
+       -- * Internals
+       , getCabalCategoriesImpl
+       , repoCabalUrl
+       ) where
+
+import Data.Text (splitOn, strip)
+import Distribution.PackageDescription
+import Distribution.PackageDescription.Parsec (parseGenericPackageDescriptionMaybe)
+
+import IW.App (App)
+import IW.Core.Repo (RepoOwner (..), RepoName (..), Category (..))
+import IW.Core.Url (Url (..))
+import IW.Effects.Download (MonadDownload (..))
+
+
+-- | Describes a monad that returns @[Category]@ given a @RepoOwner@ and @RepoName@.
+class Monad m => MonadCabal m where
+    getCabalCategories :: RepoOwner -> RepoName -> m [Category]
```

Comment:
Maybe we can somehow mention which errors we can expect from this function? (cc @chshersh )
Because it's using `MonadDownload` we can say this explicitly. What do you think?

Reply:
@vrom911 Sounds good. Should I just add logging and errors to this PR? I could do it right now

Reply:
@vrom911 I think for now I'll just add comments specifying the types of errors that may be thrown.

Reply:
It would be nice to specify errors on type-level. Though, I'm not aware of a convenient way to do so... I guess having just a comment should work for now :+1:

Reply:
@chshersh yes for sure. I will add types for errors as soon as we resolve this discussion: 
#110

Reply:
@vrom911 @chshersh The errors for parsing the cabal package description are defined here:
https://hackage.haskell.org/package/Cabal-2.4.1.0/docs/Distribution-PackageDescription-Parsec.html#t:ParseResult
How do we integrate this into our code? For now, we are using the @Maybe@ version of the parse function which returns @Nothing@ on unsuccessful parse, but we want helpful error right?

Reply:
@vrom911 @chshersh Also refer to #102 for more discussion.

Reply:
@rashadg1030 As I understand, #102 is about using something similar to `Either` instead of `Maybe` to tell specific parsing error. For now we can just tell that file failed to parse and improve the text of the message later.